### ISSUE TITLE
Reflection Probes bake incorrectly when no HDR values are visible when baked

### ISSUE THREAD

**Issue by Raidriar796** [Yellow-Dog-Man/Resonite-Issues#1851]
### Describe the bug?

The colorspace of a reflection probe is incorrect when baked in a scene where HDR values are not present or visible to the probe. This also results in the last mipmap that's generated to be a corrupted green visual for a specific mipmap step.

### To Reproduce

In a gridspace with the direction light disabled (or any world that's not bright):
1. create/inspect a reflection probe
2. press the bake reflection button or use the bake reflection probe protoflux node
3. see the debug visual is much darker then it should be
4. introduce HDR/Bloom into the scene (set light intensity above 1, spawn a bright unlit object, etc)
5. rebake the probe
6. see the debug visual looks correct

### Expected behavior

Reflection probes should bake a consistent cubemap regardless of if there are HDR values captured by the bake or not.

### Screenshots

[ReflectionProbeIssue.webm](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/145519075/4ff7def0-8367-4ecf-b507-1e5e64eaa028)

Here's a longer video showing more details of the issue, it's uploaded to streamable because it's too big to upload to Github: https://streamable.com/25ldg7

### Resonite Version Number

Beta 2024.4.26.10

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Quest 2 (Windows), Desktop (Windows/Proton)

### Log Files

[2024.4.26.10 - 2024-04-25 19_01_51.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15124764/2024.4.26.10.-.2024-04-25.19_01_51.log)

### Additional Context

This has been going on since Resonite came out but I have only just recently figured out a potential cause.

This was found and replicated with Linux using Proton but @bluecyro has confirmed it by replicating it on Windows.

A couple of things that seem to be caused by the issue:
- The incorrectly baked probe is BC1 instead of BC6H
- Incorrectly baked probes can be mostly corrected by forcing them to be in linear space or by forcing them to use the BC6H format
- The green corrupted mipmap seems to be at a consistent resolution, but I'm not quite sure what resolution the corrupted mip occurs on, but it's always the last step. (it occurs earlier on lower resolutions and later in higher resolutions)
- When a reflection probe is first created, it will be correct regardless of the lighting conditions

### Reporters

@Raidriar796 / raidriar796, @bluecyro / bluecyro

**Comment by Raidriar796** [comment]
> does the issue also occur if you start with an HDR object in the scene prior to the first bake

Yes. Incorrect probes only begin to occur when using the bake button on the component or when using protoflux nodes for baking probes. I also want to highlight that when creating a probe, the automatic bake that runs on creation is always correct.

> Does the issue persist if all HDR objects are removed from the scene?

Yes, the issue comes from there being no HDR objects in the scene/bake.

After messing around with this issue more and it seems like it's more of an issue of incorrect formats being assumed. When forcing probes to use the BC6H format they always look correct regardless of the condition they were baked in. I believe it's baking HDR by default but has sRGB applied when no HDR values are captured, and when BC1 is applied it may be applying sRGB to the image again.

**Comment by shiftyscales** [comment]
Thanks for providing additional context, and testing around this issue. :)

**Comment by Uruloke** [comment]
I've noticed I got this in my world when baking reflection probes. If I increased resolution one step from say 128 to 256 it would suddenly bake correctly into BC6H.

**Comment by shiftyscales** [comment]
@Uruloke - Even at 128 it seems to produce a BC6H variant right away.

@Raidriar796 - I followed the steps listed in this issue, but I don't seem to be able to replicate it?

Is this still an issue?

**Comment by Raidriar796** [comment]
Yes, I was able to replicate it again. 

As a quick check I opened a grid space and created a probe and rebaked it, and it was correctly BC6H. I turned off the directional light in the world and rebaked it again, and it was BC6H. I tried different resolutions and other cases where it would bake incorrectly and it was all BC6H, until I disabled the HDR checkbox. Once I baked the probe with HDR disabled it began only producing incorrectly colored BC1 probes again. 

To be more specific the original issue as initially described seems to not be an issue anymore, but SDR probe baking seem to be the actual source of the issue. Would opening a different issue be preferred for specifically targeting SDR probes and/or disabling the HDR checkbox?

**Comment by shiftyscales** [comment]
If we are unable to replicate the original issue, I'd suggest we close this, and make a new issue to focus on the additional issue / case that you discovered then, @Raidriar796.

**Comment by Uruloke** [comment]
I am still able to replicate this. Just tested one of my problematic probes in my world which I had to increase to 256 resolution to bake properly. If i return it back down to the default 128 resolution it bakes as BC1 and is really dark even though HDR is checked.

**Comment by shiftyscales** [comment]
Can yo

**Comment by Uruloke** [comment]
It's a private world hosted on my headless. I'd be happy to invite whenever I'm on but I'm not comfortable publishing the world unfortunately.

The probes will fail to bake properly depending on where they are placed and where lights are in the world.

**Comment by shiftyscales** [comment]
Could you isolate just the probes into a gridspace or similar variant of the world and see if the same issue occurs, @Uruloke?

**Comment by Uruloke** [comment]
Yeah I could try that sure.

**Comment by Uruloke** [comment]
Okay so a bit of an update. I tried to make a world that would reproduce the issue but have been unable to get it to reproduce outside of the world hosted by my headless. Even running the same world as the headless but hosted by me instead seems to fix the issue so I'm not sure what to make of it at this point.

So tldr seems to be that it currently can only be reproduced in my world running on a headless.

**Comment by Uruloke** [comment]
I tried it on my reduced world running on the headless but that didn't reproduce it either. I dunno if I removed too many things or what happened but it seems to only happen on this specific world.

**Comment by shiftyscales** [comment]
If you save a copy of the world, does the issue happen on that copy of the world as well, @Uruloke?

**Comment by Uruloke** [comment]
I do have a few copies of the world my headless can access. I'll check tomorrow.

**Comment by Uruloke** [comment]
Sorry about the delay. Yes it does indeed happen in a copy of the world running on the headless. Here's a screenshot of a probe I just baked. As you can see HDR is checked yet the probe ends up as BC1 and the sphere above shows a very dark cubemap generated.

![Image](https://github.com/user-attachments/assets/0f8683eb-2d6b-4e21-8062-c8d4924a77a8)

**Comment by ModernBalloonie** [comment]
Adding onto this issue, I experienced this issue a lot during the making of Critter Club! I can provide a replication case in a bit.

**Comment by ProbablePrime** [comment]
I'm going to fix this up to ensure we bake as HDR when HDR is checked.

Probably shortly after #1983

**Comment by ProbablePrime** [comment]
I'm investigating this one right now. I'm investigating a hunch!

However, It does touch Unity somewhat, so this might have to wait until after the Splittening. But I'm not 100% sure yet. Will update soon.

**Comment by Zyro1331** [comment]
It seems that there's some sort of Gamma filter being applied incorrectly on those baked images.
Since the darker images can be easily corrected by setting it's gamma to `0.4545`, which means somewhere it's getting corrected to `2.2` when it's not necessary.

**Comment by ProbablePrime** [comment]
That really helps thank you.

**Comment by ModernBalloonie** [comment]
<img width="1479" height="995" alt="Image" src="https://github.com/user-attachments/assets/7aed85c5-53c0-4ef1-bb10-e7beaafb0aa7" />

Interestingly, in the splittening with the recent build (Beta 2025.7.12.1268), the darkness seems to have been fixed? It's also fixed the reflection probes in Critter club. (we had a problem where probes were too dark sometimes but this update seems to have brightened them up).
